x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -750,6 +751,12 @@ private void cacheEvictionListener(RemovalNotification<String, ModelAndConsumer>
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
+        if (event.changedCustomMetadataSet().contains(TrainedModelCacheMetadata.NAME)) {
+            // Flush all models cache since we are detecting some changes.
+            logger.trace("Trained model cache invalidated on node [{}]", event.state().nodes().getLocalNodeId());
```

Comment:
nit: use a supplier or `logger.isTraceEnabled()` and avoid the call to `event.state().nodes().getLocalNodeId()` when trace is not enabled 

```suggestion
            logger.trace("Trained model cache invalidated on node [{}]", () -> event.state().nodes().getLocalNodeId());
```